rpc/types_transaction.go

```diff
@@ -304,53 +268,19 @@ func (v *TransactionVersion) BigInt() (*big.Int, error) {
 	}
 }
 
-type BroadcastedTransaction interface{}
-
-type BroadcastedTxnCommonProperties struct {
-	MaxFee *felt.Felt `json:"max_fee"`
-	// Version of the transaction scheme, should be set to 0 or 1
-	Version TransactionVersion `json:"version"`
-	// Signature
-	Signature []*felt.Felt    `json:"signature"`
-	Nonce     *felt.Felt      `json:"nonce"`
-	Type      TransactionType `json:"type"`
-}
-
-// BroadcastedInvokeV1Transaction is BROADCASTED_INVOKE_TXN
-// since we only support InvokeV1 transactions
-type BroadcastedInvokeV1Transaction struct {
-	BroadcastedTxnCommonProperties
-	SenderAddress *felt.Felt   `json:"sender_address"`
-	Calldata      []*felt.Felt `json:"calldata"`
-}
+type AddDeclareTxnInput interface{}
 
-type BroadcastedDeclareTransaction interface{}
+var _ AddDeclareTxnInput = DeclareTxnV1{}
+var _ AddDeclareTxnInput = DeclareTxnV2{}
 
-var _ BroadcastedDeclareTransaction = BroadcastedDeclareTransactionV1{}
-var _ BroadcastedDeclareTransaction = BroadcastedDeclareTransactionV2{}
+type AddDeployAccountTxnInput interface{}
```

Comment:
Why do you need this one? There is only one `DeployAccountTxn`. Also, this is only being used in `AddDeployAccountTransaction` method which could just receive a `DeployAccountTxn` as an input parameter ([spec](https://github.com/starkware-libs/starknet-specs/blob/6fadca1f1abf48523c0aa799a10061f40f973468/api/starknet_write_api.json#L152C55-L152C73))